src/Plugin.Maui.ScreenRecording/ScreenRecordingService.android.cs

```diff
@@ -2,68 +2,137 @@
-		int NOTIFICATION_ID = 1337;
-		Notification? foregroundNotification = null;
-
-		// Android O
-		if (OperatingSystem.IsAndroidVersionAtLeast(26))
-		{
-			var channelName = "Screen Recording Service";
-			var channel = new NotificationChannel(CHANNEL_ID, channelName, NotificationImportance.Default)
-			{
-				Description = "Notification Channel for Screen Recording Service"
-			};
-
-			var notificationManager = (NotificationManager?)GetSystemService(NotificationService)
-				?? throw new Exception($"{nameof(NotificationManager)} system service could not be retrieved.");
-
-			notificationManager.CreateNotificationChannel(channel);
-
-			string contentTitle = intent?.GetStringExtra("ContentTitle") ??
-				ScreenRecordingOptions.defaultAndroidNotificationTitle;
-
-			string contentText = intent?.GetStringExtra("ContentText") ??
-				ScreenRecordingOptions.defaultAndroidNotificationText;
-
-			// Create a notification for the foreground service
-			var notificationBuilder = new Notification.Builder(this, CHANNEL_ID)
-				.SetContentTitle(contentTitle)
-				.SetContentText(contentText)
-				.SetSmallIcon(Resource.Drawable.notification_template_icon_low_bg); // Ensure you have 'ic_notification' in Resources/drawable
-
-			foregroundNotification = notificationBuilder.Build();
-
-			// Start the service in the foreground
-			// Android Q
-			if (OperatingSystem.IsAndroidVersionAtLeast(29))
-			{
-				StartForeground(NOTIFICATION_ID, foregroundNotification, ForegroundService.TypeMediaProjection);
-			}
-			// Android O
-			else if (OperatingSystem.IsAndroidVersionAtLeast(26))
-			{
-				StartForeground(NOTIFICATION_ID, foregroundNotification);
-			}
-		}
-		// Any Android before O
-		else
-		{
-			StartService(intent);
-		}
-
-		return StartCommandResult.Sticky;
-	}
+    public const int NotificationId = 1337;
+    public const string ChannelId = "ScreenRecordingService";
+    public const string ExtraExternalMessenger = "ExternalMessenger";
+    public const string ExtraContentTitle = "ContentTitle";
+    public const string ExtraContentText = "ContentText";
+    public const string ExtraCommandNotificationSetup = "SetupNotification";
+    public const string ExtraCommandBeginRecording = "BeginRecording";
+    
+    public const int MsgServiceStarted = 1;
+
+    public override IBinder? OnBind(Intent? intent)
+    {
+        return null;
+    }
+
+    public override StartCommandResult OnStartCommand(Intent? intent, StartCommandFlags flags, int startId)
+    {
+        NotifyHandler(intent);
+
+        if (intent?.GetBooleanExtra(ExtraCommandNotificationSetup, false) == true)
+        {
+            SetupForegroundNotification(intent);
+        }
+
+        if (intent?.GetBooleanExtra(ExtraCommandBeginRecording, false) == true)
+        {
+            BeginRecording();
+        }
+
+        return StartCommandResult.Sticky;
+    }
+
+    public void BeginRecording()
+    {
+        try
+        {
+            // TODO ugly fix
```

Comment:
Is this still a TODO? 😅

Reply:
Yeah I think this is hackish way to do it. I was thinking of using Android's Handler to send messages from the Service.

Reply:
So will you still change that now or whats the plan?

Reply:
Unfortunately, I don't think I can update it anytime soon as I'll be very busy this week onwards. Maybe we can proceed with that for the meantime?